.github/workflows/build_and_test_usvm.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build and test USVM submodule
+
+on: 
+  push:
+    paths: 
+      - 'GameServers/usvm/'
+  pull_request:
+    paths: 
+      - 'GameServers/usvm/'
```

Comment:
Doesn't seem to work: https://github.com/PySymGym/PySymGym/pull/102

Reply:
if it indeed does not work, we may try more direct approach:
```yaml
      - name: Check for submodule changes
        run: |
          git fetch --unshallow
          old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
          new_sha=$(git rev-parse HEAD:GameServers/usvm)
          if [ "$old_sha" != "$new_sha" ]; then
            ./gradlew build
          else
            ./gradlew assemble
          fi
```

Reply:
@emnigma Looks fixed: #102

Reply:
Successful run actually **has** trailing slashes in gh-actions workflow file

<img width="651" alt="image" src="https://github.com/user-attachments/assets/13147099-c73a-447c-8d18-01ece70f3bed">

Im almost certain that check is triggered if usvm is updated **and** some changes were introduced in main repo: 

<img width="2290" alt="image" src="https://github.com/user-attachments/assets/bacf61b0-35f4-435f-8a8f-1f10861c7b61">

Reply:
Judging by ci runs, this does work (https://github.com/PySymGym/PySymGym/pull/103)

> if it indeed does not work, we may try more direct approach:
> 
> ```yaml
>       - name: Check for submodule changes
>         run: |
>           git fetch --unshallow
>           old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
>           new_sha=$(git rev-parse HEAD:GameServers/usvm)
>           if [ "$old_sha" != "$new_sha" ]; then
>             ./gradlew build
>           else
>             ./gradlew assemble
>           fi
> ```

Reply:
> if it indeed does not work, we may try more direct approach:
> 
> ```yaml
>       - name: Check for submodule changes
>         run: |
>           git fetch --unshallow
>           old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
>           new_sha=$(git rev-parse HEAD:GameServers/usvm)
>           if [ "$old_sha" != "$new_sha" ]; then
>             ./gradlew build
>           else
>             ./gradlew assemble
>           fi
> ```

This doesn't work completely: https://github.com/PySymGym/PySymGym/pull/103

CI on pull-request did launch `build` command